api/extension-guides/chat.md

```diff
@@ -0,0 +1,290 @@
+    }
+};
+```
+
+> **Tip:** Follow-ups should be written as questions or directions, not just concise commands.
+
+## Variables
+
+Orthogonal to chat participants, extensions can contribute `#variables`. Variables start with the `#` syntax and are resolved by an extension that contributes that variable, or by VS Code, if the variable is built in (e.g `#file`, `#selection`...). Users explicitly include variables in their prompts. By pressing `#`, VS Code offers a list of registered variables with their description:
+
+![List of variables in chat](images/chat/variables.png)
+
+For example, a C++ extension might contribute a variable `#cpp_context` that would get resolved based on the state of the language service - what C++ version is being used and what C++ programming approach is preferred.
+
+Variables are resolved independent of the active chat participant, and thus they can be used as a mechanism to share context between participants. For example, `@workspace` already maintains an index of the current workspace. `@workspace` contributes a variable `#codebase`, this allows users to apply the codebase context in their prompts.
+
+Variable resolvers can offer multiple length levels for the variable value. VS Code uses the one based on how many tokens are left in an LLM prompt.
+
+```typescript
+vscode.chat.registerVariable('cat_context', 'Describes the state of mind and version of the cat', {
+    resolve: (name, context, token) => {
+        if (name == 'cat_context') {
+            const mood = Math.random() > 0.5 ? 'happy' : 'grumpy';
+            return [
+                {
+                    level: vscode.ChatVariableLevel.Short,
+                    value: 'version 1.3 ' + mood
+                },
+                {
+                    level: vscode.ChatVariableLevel.Medium,
+                    value: 'I am a playful cat, version 1.3, and I am ' + mood
+                },
+                {
+                    level: vscode.ChatVariableLevel.Full,
+                    value: 'I am a playful cat, version 1.3, this version prefer to explain everything using mouse and tail metaphors. I am ' + mood
+                }
+            ]
+        }
+    }
+});
+```
+
+## Use the language model
+
+Language Model is a complementary API that allows [access to the Language Model](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.languageModels.d.ts). Using this API, if an extension contributes a chat participant it can use the Language Model to process and answer the user query. VS Code passes the exact user prompt to contributed participant, and with this API, participants can conveniently transition those language prompts into specific backend API calls. Usage of this API is not tied to the chat experience, and extensions can get language model access to empower their traditional features. For example, the Rust extension might decide to improve it's rename experience and use the Language Model to offer default names.
```

Comment:
```suggestion
Language Model is a complementary API that allows [access to the Language Model](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.languageModels.d.ts). Using this API, if an extension contributes a chat participant it can use the Language Model to process and answer the user query. VS Code passes the exact user prompt to contributed participant, and with this API, participants can conveniently transition those language prompts into specific backend API calls. Usage of this API is not tied to the chat experience, and extensions can get language model access to empower their traditional features. For example, the Rust extension might decide to improve its rename experience and use the Language Model to offer default names.
```